src/components/Dashboard/FilingHistoryList/DocumentsList.vue

```diff
@@ -31,32 +31,56 @@
 <script lang="ts">
 // Libraries
 import Vue from 'vue'
-import { Component, Emit, Prop } from 'vue-property-decorator'
-import { DocumentIF } from '@/interfaces'
+import { Component, Prop } from 'vue-property-decorator'
+import { Getter, Mutation } from 'vuex-class'
+import { ApiFilingIF, DocumentIF } from '@/interfaces'
+import { LegalServices } from '@/services'
 
 @Component({})
 export default class DocumentsList extends Vue {
   /** The filing containing documents. */
-  @Prop({ required: true }) readonly filing!: any
+  @Prop({ required: true }) readonly filing!: ApiFilingIF
 
-  /** Whether one document is currently loading. */
-  @Prop({ default: false }) readonly loadingOne!: boolean
+  @Getter getLoadingOneIndex!: number
+  @Getter isLoadingAll!: boolean
+  @Getter isLoadingOne!: boolean
 
-  /** Whether all documents are currently loading. */
-  @Prop({ default: false }) readonly loadingAll!: boolean
+  @Mutation mutateDownloadErrorDialog!: (x: boolean) => void
+  @Mutation mutateLoadingAll!: (x: boolean) => void
+  @Mutation mutateLoadingOne!: (x: boolean) => void
+  @Mutation mutateLoadingOneIndex!: (x: number) => void
```

Comment:
Is Mutations something that we'll be using moving on instead of actions? Actions will be used when there is an async call right?

Reply:
If it's only a matter of updating the store state then we can keep it simple and just call the mutations.

On the other hand, if it's more complicated then we would declare actions who in turn call the mutations.